File: BankingManagementSystem/Dto/RoleDto.cs
using System;
using System.Collections.Generic;

namespace BankingManagementSystem.Dto;

public class RoleDto
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public string ConcurrencyStamp { get; set; }
    public List<RoleClaimDto> RoleClaims { get; set; }
}

File: BankingManagementSystem.Angular/src/app/user-management/roles/roleDto.ts
import { EntityAdapter, EntityState, createEntityAdapter } from "@ngrx/entity";
import { ClaimDto } from "../claims/claimDto";

export class RoleDto{
  public id!:string;
  public name!:string;
  public concurrencyStamp!:string;
  public roleClaimsIds: string[] | [] = [];
  public roleClaims: ClaimDto[] | ClaimDto[] = [];
}

export const roleAdapter: EntityAdapter<RoleDto> = createEntityAdapter<RoleDto>({
  selectId: (role: RoleDto) => role.id,
});

export interface RolesState extends EntityState<RoleDto> {
}

export const initialRoleState: RolesState = roleAdapter.getInitialState();
